### PR TITLE
fix/ican-fee-multicurrency

### PR DESCRIPTION
**Pull request by vaidas-lungis**
icann fee multicurrency support

subtask of #2

### CODE REVIEW

**Comment by vaidas-lungis** [comment]
review @hostinger/devs

**Comment by vaidas-lungis** [comment]
Another question should we leave 0.18 hardcoded value?

File: src/Catalog/TaxIcannInterface.php [hunk 1]
```diff
@@ -7,4 +7,10 @@
     const ICANN_FEE = 0.18;
```
Comment (fordnox) [inline src/Catalog/TaxIcannInterface.php]: tai sitas kaip ir nereikalingas

File: src/Catalog/ProductDomain.php [hunk 2]
```diff
@@ -10,8 +10,20 @@ class ProductDomain extends Product implements TaxIcannInterface
 
     protected $group = Group::DOMAINS;
 
+    private $icannFee = null;
+
+    protected function getIcannValue()
+    {
+        return !is_null($this->icannFee) ? $this->icannFee : TaxIcannInterface::ICANN_FEE;
```
Comment (fordnox) [inline src/Catalog/ProductDomain.php]: jei neuzsetintas tai geriau 0 grazinam gal?
nes su multicurrency gausis nesamone

**Comment by vaidas-lungis** [comment]
@hostinger/devs updated.

**Comment by 4Giedrius** [comment]
Still amazing!

**Comment by 4Giedrius** [comment]
@vaidas-lungis I agree that branch is too amazing to be deleted! :fireworks:

File: src/CartServiceProvider.php [hunk 3]
```diff
@@ -17,6 +17,11 @@ public function register(Container $c)
             return null;
         };
 
+        $c['cart.icann_fee'] = function () use ($c) {
+            // override with your own function when registering provider
+            return null;
```
Comment (fordnox) [inline src/CartServiceProvider.php]: return 0, maybe?

File: src/Catalog/ProductDomain.php [hunk 4]
```diff
@@ -10,8 +10,21 @@ class ProductDomain extends Product implements TaxIcannInterface
...
-        return TaxIcannInterface::ICANN_FEE * $term->getPeriod();
+        return $this->icannFee * $term->getPeriod();
+    }
+
+    /**
+     * @param float $fee
+     */
+    public function setIcannFee($fee)
+    {
+        $this->icannFee = $fee;
```
Comment (fordnox) [inline src/Catalog/ProductDomain.php]: typecast to float?

**Pull request merged**